include/boost/geometry/srs/projections/impl/pj_ell_set.hpp

```diff
@@ -38,27 +42,36 @@
 #include <string>
 #include <vector>
 
+#include <boost/geometry/formulas/eccentricity_sqr.hpp>
 #include <boost/geometry/util/math.hpp>
 
-#include <boost/geometry/extensions/gis/projections/impl/pj_ellps.hpp>
-#include <boost/geometry/extensions/gis/projections/impl/pj_param.hpp>
+#include <boost/geometry/srs/projections/impl/pj_ellps.hpp>
+#include <boost/geometry/srs/projections/impl/pj_param.hpp>
+#include <boost/geometry/srs/projections/proj4.hpp>
+
 
 namespace boost { namespace geometry { namespace projections {
 
 namespace detail {
 
 /* set ellipsoid parameters a and es */
-static const double SIXTH =  .1666666666666666667; /* 1/6 */
-static const double RA4 = .04722222222222222222; /* 17/360 */
-static const double RA6 = .02215608465608465608; /* 67/3024 */
-static const double RV4 = .06944444444444444444; /* 5/72 */
-static const double RV6 = .04243827160493827160; /* 55/1296 */
+template <typename T>
+inline T SIXTH() { return .1666666666666666667; } /* 1/6 */
+template <typename T>
+inline T RA4() { return .04722222222222222222; } /* 17/360 */
+template <typename T>
+inline T RA6() { return .02215608465608465608; } /* 67/3024 */
+template <typename T>
+inline T RV4() { return .06944444444444444444; } /* 5/72 */
+template <typename T>
+inline T RV6() { return .04243827160493827160; } /* 55/1296 */
 
 /* initialize geographic shape parameters */
-inline void pj_ell_set(std::vector<pvalue>& parameters, double &a, double &es)
+template <typename BGParams, typename T>
+inline void pj_ell_set(BGParams const& bg_params, std::vector<pvalue<T> >& parameters, T &a, T &es)
 {
-    double b = 0.0;
-    double e = 0.0;
+    T b = 0.0;
```

Comment:
why not just `0`?

Reply:
Because `T` may be `double` or user-defined type, smaller fundamental types are promoted to `double` anyway.